data/field_type_enum.go

```diff
@@ -0,0 +1,156 @@
+package data
+
+// this supports the enum type
+// it is diffent than the rest since it is backed by
+// a uint16, but has special semantics and interacts with the metadata
+// Unlike the other fields it can not be easily generated
+
+// NOTE: no implementation for newEnumVectorWithValues -- since
+
+type enumVector []uint16
+
+func newEnumVector(n int) *enumVector {
```

Comment:
should there be a required argument for the list of enum variant strings?

Reply:
i don't think we can hold that at this level (private class that holds just the values) -- however, I think we can/should have a public helper that makes this more easy to do correctly.